src/main/java/com/redhat/devtools/lsp4ij/server/definition/launching/UserDefinedFormattingFeature.java

```diff
@@ -26,16 +27,80 @@
  */
 public class UserDefinedFormattingFeature extends LSPFormattingFeature {
 
+    // Server-side on-type formatting
+
     @Nullable
-    private ClientConfigurationOnTypeFormattingSettings getOnTypeFormattingSettings() {
+    private ServerSideOnTypeFormattingSettings getServerSideOnTypeFormattingSettings() {
         ClientConfigurableLanguageServerDefinition serverDefinition = (ClientConfigurableLanguageServerDefinition) getClientFeatures().getServerDefinition();
         ClientConfigurationSettings clientConfiguration = serverDefinition.getLanguageServerClientConfiguration();
-        return clientConfiguration != null ? clientConfiguration.onTypeFormatting : null;
+        return clientConfiguration != null ? clientConfiguration.format.onTypeFormatting.serverSide : null;
     }
 
     @Override
     public boolean isOnTypeFormattingEnabled(@NotNull PsiFile file) {
-        ClientConfigurationOnTypeFormattingSettings onTypeFormattingSettings = getOnTypeFormattingSettings();
-        return onTypeFormattingSettings != null ? onTypeFormattingSettings.enabled : super.isOnTypeFormattingEnabled(file);
+        ServerSideOnTypeFormattingSettings settings = getServerSideOnTypeFormattingSettings();
+        return settings != null ? settings.enabled : super.isOnTypeFormattingEnabled(file);
+    }
+
+    // Client-side on-type formatting
```

Comment:
These are all the additions for client-side on-type formatting behavior based on client config.